druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/dynamic/RocksDBDynamicLookupExtractor.java

```diff
@@ -35,6 +43,50 @@ public boolean canIterate() {
 
     @Override
     public Iterable<Map.Entry<String, String>> iterable() {
-        return staticMap.entrySet();
+        Map<String, String> tempMap = new java.util.HashMap<>();
+
+        try {
+            final RocksDB db = rocksDBManager.getDB(extractionNamespace.getNamespace());
+            RocksIterator it = db.newIterator();
+            it.seekToFirst();
+            while (it.isValid()) {
+                Optional<DynamicLookupSchema> dynamicLookupSchemaOption = schemaManager.getSchema(extractionNamespace);
+                if(!dynamicLookupSchemaOption.isPresent()) {
+                    return tempMap.entrySet();
+                }
+
+                byte[] cacheByteValue = db.get(it.key());
+                if (cacheByteValue == null) {
+                    return tempMap.entrySet();
+                }
+
+                DynamicLookupSchema dynamicLookupSchema = dynamicLookupSchemaOption.get();
+                DynamicLookupCoreSchema dynamicLookupCoreSchema = dynamicLookupSchema.getCoreSchema();
+                if (dynamicLookupCoreSchema instanceof DynamicLookupProtobufSchemaSerDe) {
+                    DynamicLookupProtobufSchemaSerDe dynamicLookupProtobufSchemaSerDe = (DynamicLookupProtobufSchemaSerDe) dynamicLookupCoreSchema;
+                    DynamicMessage dynamicMessage = Optional.of(DynamicMessage.parseFrom(dynamicLookupProtobufSchemaSerDe.getProtobufMessageDescriptor(), cacheByteValue)).get();
+                    Map<Descriptors.FieldDescriptor, Object> tempMap2 = dynamicMessage.getAllFields();
+                    StringBuilder sb = new StringBuilder();
+                    for (Map.Entry<Descriptors.FieldDescriptor, Object> kevVal: tempMap2.entrySet()) {
+                        sb.append(kevVal.getKey().getJsonName()).append(":").append(kevVal.getValue().toString()).append("#");
+                    }
+                    if (sb.length() > 0) {
+                        sb.setLength(sb.length() - 1);
+                    }
+                    StringBuilder keySb = new StringBuilder(sb);
+                    String key = keySb.substring(0, sb.indexOf("#"));
+                    tempMap.put(key, sb.toString());
+                    it.next();
+                }
+                else if (dynamicLookupCoreSchema instanceof DynamicLookupFlatbufferSchemaSerDe) {
+                    return tempMap.entrySet();
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Caught exception: " + e);
```

Comment:
pass e as argument and print error message

Reply:
Didn't get this. Can you elaborate?

Reply:
Fixed